test_metannotate_end_to_end.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+"""
+Created by: Metannotate Team (2017)
+
+Description: A shell script that runs a sample test of Metannotate and compares the generated
+             output files to verify proper end-to-end functionality of the program.
+
+Requirements: - The following packages:
+                    - build-essential
+                    - python-dev
+                    - curl
+                    - file
+                    - git
+                    - python-setuptools
+                    - ruby
+              - Base installation (sudo -H bash base_installation.sh) of Metannotate
+
+"""
+
+python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/ReferenceTest.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both
+
+echo "Verifying outputs..."
+
+cd testing/test_constants
+
+# Get reference hash file checksums
+
+FIRST_HASH=$(md5sum rpoB_0_msa_0.fa | awk '{ print $1 }')
+
+SECOND_HASH=$(md5sum rpoB_0_refseq_msa_1.fa | awk '{ print $1 }')
+
+
+cd ..
+cd ..
+
+# Store generated test output hashes in array
+
+declare -a GENERATED_FILES
+
+for entry in $(ls test_output); do
+    if [[ ${entry} == *"0_msa"* && ${entry} == *".fa"* ]]; then
+        cd test_output
+        GENERATED_FILES[0]=$(md5sum ${entry} | awk '{ print $1 }')
+        cd ..
+
```

Comment:
Drop this line. ```fi```  is the loop finisher so it should be part of the same loop. Loop finisher's were in old school languages like shell and basic have this https://en.wikipedia.org/wiki/BASIC#Structured_BASIC. Thank god for C. Fell free to put a blank line between the loops.